modular_skyrat/modules/reagent_forging/code/forge_items.dm

```diff
@@ -94,45 +119,102 @@
+
+/obj/item/forging/incomplete/shovel
+	name = "incomplete shovel head"
+	icon_state = "hot_shovelhead"
+	spawn_item = /obj/item/forging/complete/shovel
 
 //"complete" pre-complete items
+/obj/item/forging/complete
+	///the path of the item that will be created
+	var/spawning_item
+
 /obj/item/forging/complete/chain
 	name = "chain"
 	desc = "A singular chain, best used in combination with multiple chains."
 	icon_state = "chain"
 
+/obj/item/forging/complete/plate
+	name = "plate"
+	desc = "A plate, best used in combination with multiple plates."
+	icon_state = "plate"
+
 /obj/item/forging/complete/sword
 	name = "sword blade"
 	desc = "A sword blade, ready to get some wood for completion."
 	icon_state = "blade"
+	spawning_item = /obj/item/forging/reagent_weapon/sword
+
+/obj/item/forging/complete/katana
+	name = "katana blade"
+	desc = "A katana blade, ready to get some wood for completion."
+	icon_state = "katanablade"
+	spawning_item = /obj/item/forging/reagent_weapon/katana
+
+/obj/item/forging/complete/dagger
+	name = "dagger blade"
+	desc = "A dagger blade, ready to get some wood for completion."
+	icon_state = "daggerblade"
+	spawning_item = /obj/item/forging/reagent_weapon/dagger
 
 /obj/item/forging/complete/staff
 	name = "staff head"
 	desc = "A staff head, ready to get some wood for completion."
 	icon_state = "staffhead"
+	spawning_item = /obj/item/forging/reagent_weapon/staff
 
 /obj/item/forging/complete/spear
 	name = "spear head"
 	desc = "A spear head, ready to get some wood for completion."
 	icon_state = "spearhead"
+	spawning_item = /obj/item/forging/reagent_weapon/spear
+
+/obj/item/forging/complete/bokken
+	name = "bokken head"
+	desc = "A bokken head, ready to get some wood for completion."
+	icon_state = "blade"
+	spawning_item = /obj/item/forging/reagent_weapon/bokken
 
 /obj/item/forging/complete/axe
 	name = "axe head"
 	desc = "An axe head, ready to get some wood for completion."
 	icon_state = "axehead"
+	spawning_item = /obj/item/forging/reagent_weapon/axe
 
 /obj/item/forging/complete/hammer
 	name = "hammer head"
 	desc = "A hammer head, ready to get some wood for completion."
 	icon_state = "hammerhead"
-
-/obj/item/forging/complete/plate
-	name = "plate"
-	desc = "A plate, best used in combination with multiple plates."
-	icon_state = "plate"
+	spawning_item = /obj/item/forging/reagent_weapon/hammer
+
+/obj/item/forging/complete/pickaxe
+	name = "pickaxe head"
+	desc = "A pickaxe head, ready to get some wood for completion."
+	icon_state = "pickaxehead"
+	spawning_item = /obj/item/pickaxe/reagent_weapon
+
+/obj/item/forging/complete/shovel
+	name = "shovel head"
+	desc = "A shovel head, ready to get some wood for completion."
+	icon_state = "shovelhead"
+	spawning_item = /obj/item/shovel/reagent_weapon
+
+/obj/item/forging/coil
+	name = "coil"
+	desc = "A simple coil, comprised of coiled iron rods."
+	icon_state = "coil"
+
+/obj/item/stock_parts/cell/attackby(obj/item/attacking_item, mob/user, params)
```

Comment:
any cells we might wanna blacklist this from? weird cell-subtypes that we might not expect, like gun-cells or something?
might be worth checking all the subtypes of stock_parts/cell and making sure none will be problematic

Reply:
I dont think there will be any worrisome ones. It just takes the current charge and max charge of the cell.